src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipServiceTest.java

```diff
@@ -61,16 +64,18 @@ void givenDto_whenCreateLP_thenLPCreatedWithSubmissionIdAndTransactionUpdated()
         when(mapper.dtoToDao(limitedPartnershipSubmissionDto)).thenReturn(limitedPartnershipSubmissionDao);
         when(repository.insert(limitedPartnershipSubmissionDao)).thenReturn(limitedPartnershipSubmissionDao);
 
-        Transaction transaction = buildTransaction();
-
         // when
-        String submissionId = service.createLimitedPartnership(transaction, limitedPartnershipSubmissionDto, REQUEST_ID, USER_ID);
+        String submissionId = service.createLimitedPartnership(buildTransaction(),limitedPartnershipSubmissionDto, REQUEST_ID, USER_ID);
+        Transaction transaction = buildTransaction();
 
         // then
         verify(mapper, times(1)).dtoToDao(limitedPartnershipSubmissionDto);
         verify(repository, times(1)).insert(limitedPartnershipSubmissionDao);
```

Comment:
As well as verifying the call to `insert` you need to verify the call now to `save`.

Reply:
insert, save and captor verification added in test

Reply:
You need to now use the captor to validate that the link set on the submission is correct. See line 84 for an example of how that's working for the transaction captor.

Reply:
using captor to validate links in submission

Reply:
You were nearly there but now the checks have moved to a new test method.

Reply:
captor, insert and save checks returned to old test method